lib/Dialect/FIRRTL/Transforms/WireDFT.cpp

```diff
@@ -215,9 +215,48 @@ void WireDFTPass::runOnOperation() {
       });
 
   // If there are no clock gates under the DUT, we can stop now.
-  if (!clockGates.size())
+  if (clockGates.empty())
     return;
 
+  // Stash UInt<1> type for use throughout.
+  auto uint1Type = enableSignal.getType().cast<FIRRTLType>();
+
+  // Hard coded port result number; the clock gate test_en port is 1.
+  // Language below reflects this as well.
+  const unsigned testEnPortNo = 1;
+
+  // Scan gathered clock gates and check for basic compatibility.
+  for (auto *cgnode : clockGates) {
+    auto module = cgnode->getTarget()->getModule();
+    auto genErr = [&]() { return module.emitError("clock gate module "); };
+    FExtModuleOp ext = dyn_cast<FExtModuleOp>(module.getOperation());
+    if (!ext) {
+      genErr() << "must be an extmodule";
+      return signalPassFailure();
+    }
+    static_assert(testEnPortNo == 1, "update this code");
+    if (ext.getNumPorts() <= testEnPortNo) {
+      genErr() << "must have at least two ports";
```

Comment:
I think a slightly more informative error message like the one below might help `must have second port with type UInt<1>"`